### PR TITLE
iosevka: fix Darwin build

### PR DESCRIPTION
**Pull request by veehaitch**
###### Description of changes

Fixes building `iosevka` on Darwin using `buildNpmPackage` (#213481)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build iosevka iosevka.passthru.tests

**Comment by winterqt** [comment]
@wegank Was OfBorg not building it? Looks like it would have been based on the review requests...

**Comment by wegank** [comment]
There was a (network?) error on x86_64-linux:

```
trying https://nodejs.org/dist/v18.14.0/node-v18.14.0.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 18 39.5M   18 7496k    0     0  64584      0  0:10:41  0:01:58  0:08:43 85388
curl: (92) HTTP/2 stream 0 was not closed cleanly: INTERNAL_ERROR (err 2)
error: cannot download node-v18.14.0.tar.xz from any mirror
error: builder for '/nix/store/65vfq3q9abawhq1q59ay4asg7v1kipq2-node-v18.14.0.tar.xz.drv' failed with exit code 1;
```

File: pkgs/data/fonts/iosevka/default.nix [hunk 1]
```diff
@@ -67,7 +66,13 @@ buildNpmPackage rec {
 
   npmDepsHash = "sha256-Ncf07ggyOnz/2SpgdmaYS2X/8Bad+J2sz8Yyx9Iri3E=";
 
-  nativeBuildInputs = [ nodejs remarshal ttfautohint-nox ];
```
Comment (winterqt) [inline pkgs/data/fonts/iosevka/default.nix:70]: As the removal of `nodejs` isn't required for fixing Darwin, wonder if it's worth splitting into its own commit...? (cc @wegank, what do you think?)

Comment (veehaitch) [reply pkgs/data/fonts/iosevka/default.nix:70]: Done.

Comment (DamienCassou) [reply pkgs/data/fonts/iosevka/default.nix:70]: I think splitting into atomic commits is most of the times a very good idea! Thank you @veehaitch for doing it.

**Review by DamienCassou — APPROVED** [review]
Great job, thank you!

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 214344` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>iosevka</li>
    <li>iosevka-comfy.comfy</li>
    <li>iosevka-comfy.comfy-duo</li>
    <li>iosevka-comfy.comfy-fixed</li>
    <li>iosevka-comfy.comfy-motion</li>
    <li>iosevka-comfy.comfy-motion-duo</li>
    <li>iosevka-comfy.comfy-motion-fixed</li>
    <li>iosevka-comfy.comfy-wide</li>
    <li>iosevka-comfy.comfy-wide-duo</li>
    <li>iosevka-comfy.comfy-wide-fixed</li>
  </ul>
</details>

**Comment by DamienCassou** [comment]
The darwin builds don't seem to do anything. Should we wait some more?

**Comment by wegank** [comment]
@ofborg build iosevka iosevka.passthru.tests

**Pull request merged**